lib/index.js

```diff
@@ -52,9 +52,26 @@ bedrock.events.on('bedrock-express.configure.routes', app => {
     asyncHandler(async (req, res) => {
       const {type} = req.params;
       const {
-        account, email, hash, clientId, serviceId,
+        account, email, hash, serviceId,
         authenticationMethod, requiredAuthenticationMethods, typeOptions
       } = req.body;
+      let {clientId} = req.body;
+      if(!clientId) {
+        const {name} = cfg.cookies.clientId;
+        clientId = req.cookies[name];
+      }
+      if(authenticationMethod === 'token-client-registration') {
+        // generate a client id
```

Comment:
```suggestion
        // generate a client ID if one has not been set yet; otherwise
        // reuse the existing one, which improves the UX for users on
        // shared devices (each has to authenticate the client ID but
        // once authenticated it is not deleted from the device when
        // another user authenticates)
```

Reply:
Presuming this is why we're reusing an old client ID ... I've added the above comment.